clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -0,0 +1,101 @@
+
+/**
+ * A Kafka based client for sending commands.
+ *
+ * @see <a href="https://www.eclipse.org/hono/docs/api/command-and-control-kafka/">
+ *      Command &amp; Control API for Kafka Specification</a>
+ */
+public class KafkaBasedCommandSender extends AbstractKafkaBasedMessageSender implements CommandSender {
+
+    /**
+     * Creates a new Kafka-based command sender.
+     *
+     * @param producerFactory The factory to use for creating Kafka producers.
+     * @param config          The Kafka producer configuration properties to use.
+     * @param tracer          The OpenTracing tracer.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     */
+    public KafkaBasedCommandSender(
+            final KafkaProducerFactory<String, Buffer> producerFactory,
+            final KafkaProducerConfigProperties config,
+            final Tracer tracer) {
+        super(producerFactory, "command-sender", config, tracer);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * <p>
+     * The replyId is not used in the Kafka based implementation. It can be set to {@code null}.
+     * If set it will be ignored.
+     *
+     * @throws NullPointerException if tenantId, deviceId, command or correlationId is {@code null}.
+     */
+    @Override
+    public Future<Void> sendAsyncCommand(
+            final String tenantId,
+            final String deviceId,
+            final String command,
+            final String contentType,
+            final Buffer data,
+            final String correlationId,
+            final String replyId,
+            final Map<String, Object> properties,
+            final SpanContext context) {
+        Objects.requireNonNull(tenantId);
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(command);
+        Objects.requireNonNull(correlationId);
+
+        final HonoTopic topic = new HonoTopic(HonoTopic.Type.COMMAND, tenantId);
+        final Map<String, Object> headerProperties = getHeaderProperties(deviceId, command, contentType, correlationId,
+                properties);
+        return sendAndWaitForOutcome(topic.toString(), tenantId, deviceId, data, headerProperties, context);
+    }
+
+    private Map<String, Object> getHeaderProperties(final String deviceId, final String subject,
+            final String contentType, final String correlationId, final Map<String, Object> properties) {
+        final Map<String, Object> props = Optional.ofNullable(properties)
+                .orElse(new HashMap<>());
+
+        props.put(MessageHelper.APP_PROPERTY_DEVICE_ID, deviceId);
+        props.put(MessageHelper.SYS_PROPERTY_SUBJECT, subject);
+        props.put(MessageHelper.SYS_PROPERTY_CONTENT_TYPE,
+                Objects.nonNull(contentType) ? contentType : MessageHelper.CONTENT_TYPE_OCTET_STREAM);
+        props.put(MessageHelper.SYS_PROPERTY_CORRELATION_ID, correlationId);
+        props.put(KafkaMessageHelper.HEADER_RESPONSE_REQUIRED, true);
```

Comment:
shouldn't this be set only if this is a request-response command?

Reply:
_HEADER_RESPONSE_REQUIRED_ will be set to _false_ when we send one way commands. In the consequent PRs, I will add methods to send one-way commands. In the already existing API (`AsyncCommandClient`) , the method `sendAsyncCommand()` is used to send only request-response commands in asynchronous manner.

Reply:
I have set it to _true_ as there are `sendAsyncCommand(..)` methods in this PR that support only _request-response_ commands. When adding _one-way_ commands in the next PRs, I wanted to get the value for _HEADER_RESPONSE_REQUIRED_ using a parameter of `getHeaderProperties()`. I have now made these changes and pushed a commit.